charts/tempo-distributed/templates/configmap-tempo.yaml

```diff
@@ -1,14 +1,27 @@
+{{- if not .Values.useExternalConfig }}
 apiVersion: v1
+{{- if eq .Values.configStorageType "Secret" }}
+kind: Secret
+{{- else }}
 kind: ConfigMap
+{{- end }}
 metadata:
-  name: {{ template "tempo.fullname" . }}
-  namespace: {{ .Release.Namespace }}
+  name: {{ tpl .Values.externalConfigSecretName . }}
   labels:
-    {{- include "tempo.distributorLabels" . | nindent 4 }}
+    {{- include "tempo.labels" (dict "ctx" .) | nindent 4 }}
+  namespace: {{ .Release.Namespace | quote }}
+{{- if eq .Values.configStorageType "Secret" }}
+data:
+  tempo-query.yaml: {{ tpl .Values.queryFrontend.query.config . | b64enc }}
+  overrides.yaml: {{ include "tempo.calculatedOverridesConfig" . | b64enc }}
+  tempo.yaml: {{ include "tempo.calculatedConfig" . | b64enc }}
+{{- else }}
 data:
   tempo-query.yaml: |
     {{- tpl .Values.queryFrontend.query.config . | nindent 4 }}
   overrides.yaml: |
```

Comment:
This is a bit weird. Don't you reload overrides.yaml automatically in tempo? If it's automatically reloaded from a file, then you shouldn't have it next to the static config, because any change to the overrides will cause the config checksum to change and restart the services for no reason.

Reply:
I don't believe we reload on config change, but I could be wrong.

Aside from Tempo, are you saying a configmap updates the running pod without restart?  I'd no idea if that's true.  Interesting.  I'd been thinking that the annotations were in place specifically to restart the pod on change, but now I wonder.

Reply:
Marty confirmed that we will reload the overrides config.  I'll come back to tidy that up.  Thanks for the pointer.

Reply:
Perhaps we could follow up in another PR though, what do you think @krajorama ?

Reply:
Sounds ok to me as long as it's not forgotten